nvflare/job_config/script_executor.py

```diff
@@ -0,0 +1,79 @@
+if torch_ok:
+    from nvflare.app_opt.pt.params_converter import NumpyToPTParamsConverter, PTToNumpyParamsConverter
+
+    DEFAULT_PARAMS_EXCHANGE_FORMAT = ExchangeFormat.PYTORCH
+else:
+    DEFAULT_PARAMS_EXCHANGE_FORMAT = ExchangeFormat.NUMPY
+
+tensorflow, tf_ok = optional_import(module="tensorflow")
+if tf_ok:
+    from nvflare.app_opt.tf.params_converter import KerasModelToNumpyParamsConverter, NumpyToKerasModelParamsConverter
+
+
+class ScriptExecutor:
+    def __init__(
+        self,
+        script: str,
+        script_args: str = "",
+        launch_external_process: bool = False,
+        params_transfer_type: TransferType = TransferType.FULL,
+        params_exchange_format=DEFAULT_PARAMS_EXCHANGE_FORMAT,
+    ):
+        """ScriptExecutor is used with FedJob API to run or launch a script.
+
+        in-process `launch_external_process=False` uses InProcessClientAPIExecutor (default).
+        ex-process `launch_external_process=True` uses ClientAPILauncherExecutor.
+
+        Args:
+            script (str): Script to run. For in-process must be a python script path. For ex-process can be any command supported by python subprocess.
+            script_args (str): Optional arguments for script.
+            launch_external_process (bool): Whether to launch the script in external process. Defaults to False.
+            params_transfer_type (str): How to transfer the parameters. FULL means the whole model parameters are sent.
+                DIFF means that only the difference is sent. Defaults to TransferType.FULL.
+            params_exchange_format (str): Format to exchange the parameters. Defaults based on detected imports.
+        """
+        self._script = script
+        self._script_args = script_args
+        self._launch_external_process = launch_external_process
+        self._params_transfer_type = params_transfer_type
+        self._params_exchange_format = params_exchange_format
+
+        self._from_nvflare_converter = None
+        self._to_nvflare_converter = None
+
+        if torch_ok:
+            if params_exchange_format == ExchangeFormat.PYTORCH:
+                self._from_nvflare_converter = NumpyToPTParamsConverter(
+                    [AppConstants.TASK_TRAIN, AppConstants.TASK_VALIDATION]
+                )
+                self._to_nvflare_converter = PTToNumpyParamsConverter(
+                    [AppConstants.TASK_TRAIN, AppConstants.TASK_SUBMIT_MODEL]
+                )
+        if tf_ok:
+            if params_exchange_format == ExchangeFormat.NUMPY:
+                self._from_nvflare_converter = NumpyToKerasModelParamsConverter(
+                    [AppConstants.TASK_TRAIN, AppConstants.TASK_VALIDATION]
+                )
+                self._to_nvflare_converter = KerasModelToNumpyParamsConverter(
+                    [AppConstants.TASK_TRAIN, AppConstants.TASK_SUBMIT_MODEL]
+                )
+        # TODO: support other params_exchange_format
```

Comment:
if no self._from_nvflare_converter and no self._to_nvflare_converter, what will happen?